File: C & C++/Cpp/whileProdotti.cpp
#include <iostream>

using namespace std;

int main() {
    int costoProdotto;
    int budget = 100;
    int prodotti = 0;

    while (true) {
        cout << "Inserisci il costo del prodotto: ";
        cin >> costoProdotto;

        if (budget < costoProdotto) {
            cout << "Non puoi acquistare questo prodotto, il tuo budget è troppo basso." << endl;
            break;
        }

        budget -= costoProdotto;
        prodotti++;
        cout << "Prodotto acquistato, Budget rimanente: " << budget << endl;

        if (budget == 0) {
            break;
        }
    }

    cout << "Hai acquistato " << prodotti << " prodotti." << endl;

    return 0;
}

File: Project/python/whileProdotti.py
costoProdotto = 0
budget = 100
prodottiAcquistati = 0

while True:
    costoProdotto = int(input("Inserisci costo prodotto: "))

    if costoProdotto > budget:
        print("Non hai abbastanza budget per acquistare questo prodotto")
        break
    else:
        budget -= costoProdotto
        prodottiAcquistati += 1
        print(f"Prodotto acquistato! Budget rimanente: {budget}")

    if budget == 0:
        break

print(f"Hai acquistato {prodottiAcquistati} prodotti")